### PR TITLE
IDENTITY-5983: Fix user password reset after tenant is deactivated

### PR DESCRIPTION
**Pull request by emswbandara**

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java [hunk 1]
```diff
@@ -143,6 +144,13 @@ public void updatePassword(String code, String password, Property[] properties)
 
         UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
         UserRecoveryData userRecoveryData = userRecoveryDataStore.load(code);
+
+        String contextTenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain() ;
+        String userTenantDomain = userRecoveryData.getUser().getTenantDomain();
+
+        if(!StringUtils.equals(contextTenantDomain, userTenantDomain)){
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Formatting issue

Comment (emswbandara) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fixed with https://github.com/wso2-extensions/identity-governance/pull/201

Comment (pulasthi7) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fix formatting

Comment (emswbandara) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fixed with #201

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java [hunk 2]
```diff
@@ -143,6 +144,13 @@ public void updatePassword(String code, String password, Property[] properties)
 
         UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
         UserRecoveryData userRecoveryData = userRecoveryDataStore.load(code);
+
+        String contextTenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain() ;
```
Comment (omindu) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Formatting issue.

Comment (emswbandara) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fixed with https://github.com/wso2-extensions/identity-governance/pull/201

**Pull request merged**